evm/src/main/java/org/hyperledger/besu/evm/processor/AbstractMessageProcessor.java

```diff
@@ -200,20 +207,12 @@ public void process(final MessageFrame frame, final OperationTracer operationTra
       }
     }
 
-    if (frame.getState() == MessageFrame.State.EXCEPTIONAL_HALT) {
-      exceptionalHalt(frame);
-    }
-
-    if (frame.getState() == MessageFrame.State.REVERT) {
-      revert(frame);
-    }
-
-    if (frame.getState() == MessageFrame.State.COMPLETED_SUCCESS) {
-      completedSuccess(frame);
-    }
-
-    if (frame.getState() == MessageFrame.State.COMPLETED_FAILED) {
-      completedFailed(frame);
+    switch (frame.getState()) {
```

Comment:
AFAIK the `switch` statement is not equivalent to the chain of `if`s. More than one if condition could be triggered, if the state changes during the execution of `exceptionalHalt()`, `revert()`, `completedSuccess()` or `completedFailed()`. I think we need to revert this change.

Reply:
Fair point.